receiver/snmpreceiver/config.go

```diff
@@ -448,18 +455,28 @@ func validateColumnOID(metricName string, columnOID ColumnOID, cfg *Config) erro
 
 	// Check that any ResourceAttributes have a valid value
 	if len(columnOID.ResourceAttributes) > 0 {
-		hasIndexedIdentifier = true
 		for _, name := range columnOID.ResourceAttributes {
-			_, ok := cfg.ResourceAttributes[name]
+			resourceAttribute, ok := cfg.ResourceAttributes[name]
 			if !ok {
 				combinedErr = multierr.Append(combinedErr, fmt.Errorf(errMsgColumnResourceAttributeBadName, metricName, name))
+				continue
+			} 
+
+			if resourceAttribute.OID != "" || resourceAttribute.IndexedValuePrefix != "" { hasIndexedIdentifier = true }
+			
+			numKeys := 0
+			if resourceAttribute.OID != "" {numKeys++}
+			if resourceAttribute.ScalarOID != "" {numKeys++}
+			if resourceAttribute.IndexedValuePrefix != "" {numKeys++}
+			if numKeys > 1 {
```

Comment:
Is there a problem if no keys are set?

Reply:
More generally, should we _only_ apply this validation here? i.e. Why not apply it in `validateResourceAttributeConfigs`? Unless it's something specific to column OIDs, I think we should move it.

Reply:
Yep, that's a great point.